libraries/AP_Scripting/applets/Heli_IM_Col_Tune.lua

```diff
@@ -0,0 +1,140 @@
+-- This script is a useful tool when first setting up a heli.  You can use it to dial in the IM_STB_COL_2 and 
+-- IM_STAB_COL_3 parameters in a more intuative way.  It is meant for use with a transmitter that has two pots
+-- available.  1 Pot can be used to control the gradient of the line between the 40% and 60% curve points.  Use 
+-- this pot to adjust the sensitivity of the collective about the collective midstick.  The 2nd Pot then controls
+-- the value of the 50% point on the curve.  This can be used to set the collective position to aid with hovering 
+-- at the midstick.
+
+-- Tested and working as of 21st Aug 2020 (Copter Dev)
+
+
+function update()
+    local rcin_50 = col50_val_ch:norm_input_ignore_trim()
+    local rcin_grad = col_grad_ch:norm_input_ignore_trim()
+
+    -- Ensure that the channel hasn't been reversed
+    if col50_val_ch:get_reverse() then
+        rcin_50 = rcin_50*-1
+    end
+    if col_grad_ch:get_reverse() then
+        rcin_grad = rcin_grad*-1
+    end
+
+    -- Offset starting midstick curve value to get new value
+    -- Max = +30% , Min = -30%
+    local im_50 = (value_im_3 + value_im_2) * 0.5 --(%)
+    local delta_50 = rcin_50*30
+    im_50 = im_50 + delta_50
+
+    -- Scale rcin_grad to be between 0 and 1
+    rcin_grad = (rcin_grad+1)*0.5
+
+    -- Calculate delta due to gradient
+    local grad_40 = rcin_grad * -10 --(%)
+    local grad_60 = rcin_grad * 10 --(%)
+
+    -- Calculate param values
+    local im_40_pct = im_50+grad_40
+    local im_60_pct = im_50+grad_60
+
+    -- Ensure IM_STB_COL_2 < IM_STB_COL_3
+    if im_40_pct >= im_60_pct then
+        im_40_pct = im_60_pct - 1
+    end
+
+    -- Enforce that IM_STB_COL_1 < IM_STB_COL_2
+    local im_0_pct = get_im_val('IM_STB_COL_1',false)
+    if im_0_pct >= im_40_pct then
+        -- Correct within parameter limits
+        im_0_pct = min_max(im_40_pct - 1)
+
+        -- Set correct value to prevent pre-arm warnings
+        if not(param:set('IM_STB_COL_1',im_0_pct)) then
+            gcs:send_text(2, 'LUA: param set failed')
+        end
+    end
+
+    -- Enforce that IM_STB_COL_4 > IM_STB_COL_3
+    im_100_pct = get_im_val('IM_STB_COL_4',false)
+    if im_100_pct <= im_60_pct then
+        -- Correct within parameter limits
+        im_100_pct = min_max(im_60_pct + 1)
+
+        -- Set correct value to prevent pre-arm warnings
+        if not(param:set('IM_STB_COL_4',im_100_pct)) then
```

Comment:
only setting the params will not be persistent you would need to set_and_save

Reply:
Arguably you would want to set it normally, and have a separate switch/interaction to set and save, but as long as people back it up first it's probably okay?

Reply:
I have added a switch that saves the params and exits the script